src/model/keychain.ts

```diff
@@ -70,6 +89,15 @@ export async function loadString(
   options?: Options
 ): Promise<null | string | -1 | -2> {
   try {
+    // if the data is public, try to load data from mmkv first
+    const data = loadStringMMKV(key);
+    if (data) {
+      console.log('data is found: ', data);
+      return data;
+    }
+    console.log('data is not found using keychain');
+
+    // if not found or private, load from keychain
```

Comment:
did we want to consider a public-data migration on load (depending on how often public data is expected to be written or not)

Reply:
was going to change this but we cant do it on load since we dont know at that point if the data is private or public. its better to ensure its not private and do it where we can safely do that

imo this is okay here since on app start up we update and save this data on app start up already